10up/classifai
# Fix Computer Vision settings when saving for the first time

### Description of the Change

This update fixes the Microsoft Computer Vision settings page checkbox options always being saved  with the value '1' if the setting had not yet been saved before.

The `sanitize_settings()` callback sets each checkbox value to:
- `no` if the option is empty (checkbox unchecked)
- `1` if the option is not empty (checkbox checked)

This logic leads to trouble when the `sanitize_settings()` callback is called twice, which happens when saving the settings for the first time. When saving a setting for the first time the `update_option()` and the `add_option()` functions are called which both run the sanitize callback.

This update also contains a minor refactor of the way the settings page error notifications are registered. It prevents the same error being rendered twice when the callback is called multiple times.

### Benefits

The settings are saved correctly when saving the options for the first time.

### Verification Process

To verify this update:

Before applying this update:

- Delete the `classifai_computer_vision` option
- Go to Image Processing settings page
- Save options
- The "Automatically Caption Images", "Automatically Tag Images" and "Enable smart cropping" checkboxes are all checked

After applying this update:
- Delete the `classifai_computer_vision` option
- Go to Image Processing settings page
- Save options
- The "Automatically Caption Images", "Automatically Tag Images" and "Enable smart cropping" checkboxes are checked or unchecked correctly
